[memgraph-2-12 <] Write docs for refactor.rename_type_property (#271)
memgraph/documentation: pages/advanced-algorithms/available-algorithms/refactor.mdx

+-------------------------------------------------------------------------------------------------------------------+
```

![](/pages/advanced-algorithms/available-algorithms/refactor/rel_after.png)

> 

Reviewer: ```suggestion
```

### `rename_type_property()`

Renames the property of a relationship.

{<h4> Input: </h4>}

> - `old_property: string` ➡  relationship property that will be renamed.
> - `new_property: string` ➡  new property name.
> - `rels: List[Relationship]` ➡  list of relationships whose properties will be renamed.
> 
> {<h4> Output: </h4>}
> 
> - `relationships_changed: integer` ➡ number of relationships whose properties were changed.

Reviewer: ```suggestion
- `old_property: string` ➡  The relationship property that will be renamed.
- `new_property: string` ➡  The new property name.
- `rels: List[Relationship]` ➡  The list of relationships whose properties will be renamed.

{<h4> Output: </h4>}

- `relationships_changed: integer` ➡ The number of relationships with a changed property name.
```

{<h4> Usage: </h4>}

> Create the following graph:

Reviewer: ```suggestion
Create a graph using the queries below:
```

```cypher
CREATE (h:Human {name:"Carl"})-[r1:RUNS {distance_in_km:20}]->(d:Destination),
(h1:Human {name:"Marta"})-[r2:RUNS {distance_in_km: 20}]->(d),
(h2:Human {name:"Peter"})-[r3:RUNS {distance_in_km: 40}]->(d);
```

> 
> Graph and relationships properties before:

Reviewer: ```suggestion
```

![](/pages/advanced-algorithms/available-algorithms/refactor/rename_prop_before.png)

> Change the name of `distance_in_km` property of all relationships:

Reviewer: ```suggestion
Use the procedure to change the name of `distance_in_km` property of all relationships into `distance`:
```

```cypher
MATCH ()-[r]-()
WITH COLLECT(distinct(r)) AS rels

> CALL refactor.rename_type_property("distance_in_km", "distance",rels) YIELD relationships_changed RETURN relationships_changed;

Reviewer: ```suggestion
CALL refactor.rename_type_property("distance_in_km", "distance", rels) YIELD relationships_changed RETURN relationships_changed;
```

```

> 

Reviewer: ```suggestion
The procedure returns the number of updated relationships:

```

```plaintext
+---------------------------------------+
| relationships_changed                 |
+---------------------------------------+
| 3                                     |
+---------------------------------------+
```

> Graph and relationships properties after:

Reviewer: ```suggestion
The graph and relationship properties after the update:
```

![](/pages/advanced-algorithms/available-algorithms/refactor/rename_prop_after.png)

> Change the name of `distance_in_km` property in a specific relationship, in this case, relationships going from `Human` node with `name` property `Carl`:

Reviewer: ```suggestion
The following query will change the name of `distance_in_km` property in a specific relationship, in this case, relationship going from the `Human` node with the `name` property `Carl`:
```
